src/main/java/org/zalando/nakadi/controller/SubscriptionController.java

```diff
@@ -143,4 +148,35 @@ private Subscription getExistingSubscription(final SubscriptionBase subscription
                 subscriptionBase.getConsumerGroup());
     }
 
+    private final class EventTypesHolder {
```

Comment:
@v1ctor I did not quite get your point.
What is 'repository.exists(type)' and why do we need it?
I think the simpler will be to use approach with subset:
        getExistingEventTypes().stream()
                .filter(eventType -> List<EventType> missingEventTypes = subscriptionBase.getEventTypes().contains(eventType.getName()))
                .map(EventType::getName)
                .collect(toList());

But I preferred to do it with the additional object because I think it is good to save additional cpu cycles.